include/gnc_environment.hpp

```diff
@@ -0,0 +1,44 @@
+//
+// include/gnc_environment.hpp
+// PSim
+//
+// Contributors:
+//   Kyle Krol  kpk63@cornell.edu
+//
+// Pathfinder for Autonomous Navigation
+// Space Systems Design Studio
+// Cornell Univeristy
+//
+
+#ifndef PAN_PSIM_INCLUDE_GNC_ENVIRONMENT_HPP_
+#define PAN_PSIM_INCLUDE_GNC_ENVIRONMENT_HPP_
+
+#include "lin.hpp"
+
+namespace gnc {
+namespace env {
+
+/** @fn earth_attitude
+ *  @param[in]  t Time in seconds since the PAN epoch.
+ *  @param[out] q Quaternion to transform vectors from ECI to ECEF. */
+void earth_attitude(double t, lin::Vector4d &q);
+
+/** @fn earth_attitude
+ *  @param[in]  t Time in seconds since the PAN epoch.
+ *  @param[out] q Quaternion to transform vectors from ECI to ECEF. */
+void earth_attitude(double t, lin::Vector4f &q);
```

Comment:
What's the utility of these float-based functions? Wouldn't it be better to do all of the arithmetic with doubles, and then convert to floats only if necessary for downlinking?

Reply:
The only thing would just be wasting instructions if we use doubles all the time (although that may not be a huge deal). For the moment, I've been trying to to specialize because we aren't exactly sure what the controllers are going to look like.

For example, I know we want to do attitude stuff in floats because we don't need the accuracy. However, the time here clearly needs to be passed as a double for precision so we can be multiplying that double onto a float before normalizing the quaternion. However, once the quaternion is normalized our precision issues disappear. So in that case, the float function has a utility.

Reply:
Yeah I suppose it's not a huge deal to have them, either, but eventually we want to be careful about having too many functions to deal with

Reply:
Internally we will have a lot of functions most likely. But the interface with the flight computer will hopefully be very narrow.